x-pack/plugins/index_management/public/application/components/mappings_editor/components/document_fields/fields/create_field/semantic_text/use_semantic_text.ts

```diff
@@ -126,49 +153,36 @@ export function useSemanticText(props: UseSemanticTextProps) {
-      try {
-        const modelDownloaded: boolean = await isModelDownloaded(trainedModelId);
-
-        if (isDeployed) {
-          createInferenceEndpoint(trainedModelId, defaultInferenceEndpoint, data);
-        } else if (modelDownloaded) {
-          ml?.mlApi?.trainedModels
-            .startModelAllocation(trainedModelId)
-            .then(() => createInferenceEndpoint(trainedModelId, defaultInferenceEndpoint, data));
-        } else {
-          ml?.mlApi?.trainedModels
-            .installElasticTrainedModelConfig(trainedModelId)
-            .then(() => ml?.mlApi?.trainedModels.startModelAllocation(trainedModelId))
-            .then(() => createInferenceEndpoint(trainedModelId, defaultInferenceEndpoint, data));
+    const { trainedModelId, isDeployed, isDeployable } = inferenceData;
+    // trainedModelId is empty string when it's a third party model
+    try {
+      if (!trainedModelId) {
+        await createInferenceEndpoint(trainedModelId, data, customInferenceEndpointConfig);
+      } else {
+        if (isDeployable) {
+          const modelDownloaded: boolean = await isModelDownloaded(trainedModelId);
+          if (isDeployed) {
+            await createInferenceEndpoint(trainedModelId, data, customInferenceEndpointConfig);
+          } else if (modelDownloaded) {
+            ml?.mlApi?.trainedModels
+              .startModelAllocation(trainedModelId)
+              .then(() =>
+                createInferenceEndpoint(trainedModelId, data, customInferenceEndpointConfig)
+              );
+          } else {
+            ml?.mlApi?.trainedModels
+              .installElasticTrainedModelConfig(trainedModelId)
+              .then(() => ml?.mlApi?.trainedModels.startModelAllocation(trainedModelId))
+              .then(() => createInferenceEndpoint(trainedModelId, data));
+          }
+          NotificationToasts({ toasts });
         }
-        toasts?.addSuccess({
-          title: i18n.translate(
-            'xpack.idxMgmt.mappingsEditor.createField.modelDeploymentStartedNotification',
-            {
-              defaultMessage: 'Model deployment started',
-            }
-          ),
-          text: i18n.translate(
-            'xpack.idxMgmt.mappingsEditor.createField.modelDeploymentNotification',
-            {
-              defaultMessage: '1 model is being deployed on your ml_node.',
-            }
-          ),
-        });
-      } catch (error) {
+      }
+    } catch (error) {
+      if (trainedModelId) {
         setErrorsInTrainedModelDeployment?.((prevItems) => [...prevItems, trainedModelId]);
-        toasts?.addError(error.body && error.body.message ? new Error(error.body.message) : error, {
-          title: i18n.translate(
-            'xpack.idxMgmt.mappingsEditor.createField.modelDeploymentErrorTitle',
-            {
-              defaultMessage: 'Model deployment failed',
-            }
-          ),
-        });
       }
+      NotificationToasts({ toasts, error });
```

Comment:
Same here: we know this is an error, so we can just directly show the error toast